pkg/auth/webapp/result.go

```diff
@@ -45,6 +45,11 @@ func (r *Result) WriteResponse(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	// Propagate q_from
+	if req.URL.Query().Get(QueryFrom) != "" {
+		q.Set(QueryFrom, req.URL.Query().Get(QueryFrom))
+	}
```

Comment:
Wouldn't this make `q_from` like `x_from`?

Reply:
We only propagate q_from in an redirect response, but not in any page.
Rationale: When user POST something to an endpoint, and that endpoint returned an redirect response. In the user's perspective the user is reaching the redirected uri from the page he POST the request.

Reply:
Added a comment to explain it